tests/shared/drive-constraints.spec.js

```diff
@@ -254,50 +430,140 @@ describe('Shared: DriveConstraints', function() {
 
   describe('.isDriveValid()', function() {
 
-    describe('given drive is large enough', function() {
-
-      beforeEach(function() {
-        this.drive = {
-          device: '/dev/disk2',
-          name: 'My Drive',
-          size: 4000000000
-        };
-        this.image = {
-          path: 'rpi.img',
-          size: 2000000000
-        };
-      });
+    it('should return false if the drive is not large enough, is locked, and is a source drive', function() {
```

Comment:
Is there any way these eight separate tests (i.e. all the combinations of three separate conditions) could be run in some kind of auto-generated loop, rather than laboriously being typed out individually? Obviously if we added another couple of extra conditions to "what makes a drive valid" we wouldn't really want to have to type out 32 individual tests.